### PR TITLE
 Improving test coverage for databridge.agent and databride.commons.binary

### PR DESCRIPTION
**Pull request by pcnfernando**

### CODE REVIEW

File: components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Datapublisher Test Case.
+ */
+public class DataPublisherTestCase {
+    Logger log = Logger.getLogger(OneEndPointDPBinaryTest.class);
```
Comment (tishan89) [inline components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java]: Access modifier is missing

File: components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+        int numberOfEventsSent = 1000;
+        for (int i = 0; i < numberOfEventsSent; i++) {
+            dataPublisher.tryPublish(streamID, metaData, correlationData, payLoad, arbitrary);
+        }
+        try {
+            Thread.sleep(5000);
+        } catch (InterruptedException e) {
+        }
+        dataPublisher.shutdown();
```
Comment (tishan89) [inline components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java]: Where are the asserts? Applies to all other testcases as well

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/java/org/wso2/carbon/databridge/commons/binary/test/BinaryMessageConvertUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetEncoder;
+
+
+/**
+ * Binary Message Convert Util Testcase.
+ */
+public class BinaryMessageConvertUtilTest {
+    Logger log = Logger.getLogger(BinaryMessageConvertUtilTest.class);
```
Comment (tishan89) [inline components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/java/org/wso2/carbon/databridge/commons/binary/test/BinaryMessageConvertUtilTest.java]: Missing access modifier

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (tishan89) [inline components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties]: Use WSO2 licence

File: components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java (hunk 1 above)
Comment (pcnfernando) [reply components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java]: Fixed with commit [#1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a](https://github.com/pcnfernando/carbon-analytics-common/commit/1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a)

File: components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java (hunk 2 above)
Comment (pcnfernando) [reply components/data-bridge/org.wso2.carbon.databridge.agent/src/test/java/org/wso2/carbon/databridge/agent/test/DataPublisherTestCase.java]: Fixed with commit [#1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a](https://github.com/pcnfernando/carbon-analytics-common/commit/1ab4f1d9e9b2c37d7266d546f4c09b775b4af81a)

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# has a default priority. All other categories need not be assigned a
+# priority in which case they inherit their priority from the
+# hierarchy.
+
+#log4j.rootLogger=debug, console
+log4j.rootLogger=info, console
+
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+#log4j.appender.console.layout.ConversionPattern=[%t] %-5p %c %x - %m%n
```
Comment (dakshika) [inline components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties]: Remove unwanted comments, same appender on the next line

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+# documenation of org.apache.log4j.PropertyConfigurator.
+
+# The root category uses the appender called A1. Since no priority is
+# specified, the root category assumes the default priority for root
+# which is DEBUG in log4j. The root category is the only category that
+# has a default priority. All other categories need not be assigned a
+# priority in which case they inherit their priority from the
+# hierarchy.
+
+#log4j.rootLogger=debug, console
```
Comment (dakshika) [inline components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties]: Remove unwanted comment same property uncommented on next line

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties (hunk 5 above)
Comment (pcnfernando) [reply components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties]: Fixed in commit [#400769d9711a574a08ea10faf5cd99ccd2edbc78](https://github.com/wso2/carbon-analytics-common/commit/400769d9711a574a08ea10faf5cd99ccd2edbc78)

File: components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties (hunk 6 above)
Comment (pcnfernando) [reply components/data-bridge/org.wso2.carbon.databridge.commons.binary/src/test/resources/log4j.properties]: Fixing with this PR [#553](https://github.com/wso2/carbon-analytics-common/pull/553)

**Pull request merged**